widget/button.go

```diff
@@ -221,6 +231,57 @@ func (b *Button) Tapped(*fyne.PointEvent) {
 	}
 }
 
+// TappedSecondary is called when a secondary pointer tapped event is captured
+func (b *Button) TappedSecondary(*fyne.PointEvent) {
+}
+
+// FocusGained is called when the Entry has been given focus.
+func (b *Button) FocusGained() {
+	if b.Disabled() {
+		return
+	}
+	b.focused = true
+	b.Refresh()
+}
+
+// FocusLost is called when the Entry has had focus removed.
+func (b *Button) FocusLost() {
+	b.focused = false
+	b.Refresh()
+}
+
+// Focused returns whether or not this Entry has focus.
+func (b *Button) Focused() bool {
+	return b.focused
+}
+
+// TypedRune receives text input events when the Check is focused.
+func (b *Button) TypedRune(r rune) {
+}
+
+func (b *Button) TypedKey(key *fyne.KeyEvent) {
+	if b.Disabled() {
+		return
+	}
+	if key.Name == fyne.KeyReturn || key.Name == fyne.KeyEnter || key.Name == fyne.KeySpace {
```

Comment:
This logic is used multiple times - perhaps it should be de-duplicated?

Reply:
Perhaps. But I do not see any good way to do it.

Reply:
A private function called ‘isSubmitKey(key)’ perhaps?